### ISSUE TITLE
IsMatch throws IndexOutOfRangeException

### ISSUE THREAD

**Issue by sualitu** [dazinator/DotNet.Glob#44]
If I run:

```
var file = "x";
var glob = "**/y"
Glob.Parse(glob).IsMatch(file);
```

The following exception is thrown by `IsMatch`

```
Unhandled Exception: System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at DotNet.Globbing.Evaluation.WildcardDirectoryTokenEvaluator.IsMatch(String allChars, Int32 currentPosition, Int32& newPosition)
   at DotNet.Globbing.Evaluation.CompositeTokenEvaluator.IsMatch(String allChars, Int32 currentPosition, Int32& newPosition)
   at DotNet.Globbing.Glob.IsMatch(String subject)
```

It seems that if the length of the string to match is the same as what follows the `**/` the exception is thrown.

Additional information:
```
λ dotnet --version
2.1.4
```

**Comment by dazinator** [comment]
may I ask what version of dotnet.glob nuget package you are using?

**Comment by sualitu** [comment]
Of course you may: `<PackageReference Include="DotNet.Glob" Version="2.0.0" />`

**Comment by dazinator** [comment]
Thanks, I will add a test case for this and hopefully, it will reproduce, in which case I'll get a fix out.
Thanks for taking the time to report :-)

**Comment by dazinator** [comment]
I have fixed this in version `2.0.1` which has been uploaded to nuget. Thank you for finding this edge case :-)
